lms/services/course.py

```diff
@@ -114,7 +114,25 @@ def search(  # noqa: PLR0913, PLR0917
                 .filter(User.h_userid == h_userid)
             )
 
-        return query.limit(limit).all()
+        return query.limit(limit)
+
+    def search(self, **kwargs) -> list[Course]:
+        return self._search_query(**kwargs).all()
```

Comment:
Would `kwargs` here allow to get IDE intellisense, and statically warn you if you introduce a typo (like `id=123` instead of `id_=123`) or make some other human error?

If it doesn't, maybe I would suggest explicitly defining the same arguments as in `_search_query`, even if it's a bit more verbose.

Reply:
Yeah, you are right, added the full method signature in search :+1: